index.py

```diff
@@ -154,12 +163,41 @@ def get_documents(self):
             document['date_posted_positional_indexes'] = self.generate_positional_indexes(document['date_posted'].split()[0])  # Part 4: Date_posted
             set_of_documents.append(document)
 
+            # To obtain the top K terms for the current document
+            document['top_K'] = []
+            accumulate_counts = {}
+            self.include_count_contribution_from_pos_ind(accumulate_counts, document['content_positional_indexes'])
+            self.include_count_contribution_from_pos_ind(accumulate_counts, document['title_positional_indexes'])
+            self.include_count_contribution_from_pos_ind(accumulate_counts, document['court_positional_indexes'])
+            self.include_count_contribution_from_pos_ind(accumulate_counts, document['date_posted_positional_indexes'])
+            document['top_K'] = Counter(accumulate_counts).most_common(K)
+            for i in range(K):
+                # i must always be smaller than actual_size by 1
+                # accumulate_counts has a possibility of going below K
+                # to avoid null pointer exception, we use < len(accumulate_counts)
+                if (i < len(accumulate_counts)):
+                    document['top_K'].append(document['top_K'][i][0])
+                else:
+                    break;
```

Comment:
Why append here though. If you do this you are appending duplicate data, unless this is what you intended

Reply:
:o sry oversight!!!

Reply:
Changed to 
                    document['top_K'][i] = document['top_K'][i][0]

Reply:
Chaging it to `document['topK'] = [doc[0] for doc in document['topK'][:accumulate_counts)] would be clearer

Reply:
Get rid of the loop

Reply:
changed to 
            document['top_K'] = [doc[0] for doc in document['top_K'][:accumulate_counts]]

Reply:
ehhh ^ that's wrong tho, there's now an error.
Reverting to 
for i in range(K):
                # i must always be smaller than actual_size by 1
                # accumulate_counts has a possibility of going below K
                # to avoid null pointer exception, we use < len(accumulate_counts)
                if (i < len(accumulate_counts)):
                    document['top_K'].append(document['top_K'][i][0])
                else:
                    break;